### PR TITLE
Fix handling of iterable fields

### LINKED ISSUE #196: `column_formatters` doesn't work with one-to-many relation
### Checklist

- [X] The bug is reproducible against the latest release or `master`.
- [X] There are no similar issues or pull requests to fix it yet.

### Describe the bug

The details page fails to render the list value formatted by `column_formatters` and `column_formatters_detail` because the following template tries to pass it to `_url_for_details()` instead of the original model value.

https://github.com/aminalaee/sqladmin/blob/83d8d2540f8a0fdca1e23ae35e37580b6dec06e6/sqladmin/templates/details.html#L24-L26

### Steps to reproduce the bug

```python
from typing import List

from sqlalchemy import Column, ForeignKey, Integer, create_engine
from sqlalchemy.ext.declarative import declarative_base

Base = declarative_base()
engine = create_engine(...)


class User(Base):
    __tablename__ = "users"

    id = Column(Integer, primary_key=True)
    addresses: List["Address"] = relationship("Address", back_populates="user")


class Address(Base):
    __tablename__ = "addresses"

    id = Column(Integer, primary_key=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    user: User = relationship("User", back_populates="addresses")


Base.metadata.create_all(engine)
```

```python
from fastapi import FastAPI
from sqladmin import Admin, ModelAdmin

app = FastAPI()
admin = Admin(app, engine)


class UserAdmin(ModelAdmin, model=User):
    column_list = [User.id, User.address]
    column_formatters = {
        User.addresses: lambda m, a: [a.id for a in m.addresses],
    }


admin.register_model(UserAdmin)
```

### Expected behavior

_No response_

### Actual behavior

_No response_

### Debugging material

```python
  File "/myproject/.venv/lib/python3.9/site-packages/sqladmin/templates/details.html", line 25, in block 'content'
    <a href="{{ model_admin._url_for_details(value) }}">({{ value }})</a>
  File "/myproject/.venv/lib/python3.9/site-packages/sqladmin/models.py", line 567, in _url_for_details
    pk = getattr(obj, inspect(obj).mapper.primary_key[0].name)
  File "/myproject/.venv/lib/python3.9/site-packages/sqlalchemy/inspection.py", line 71, in inspect
    raise exc.NoInspectionAvailable(
sqlalchemy.exc.NoInspectionAvailable: No inspection system is available for object of type <class 'str'>
```

### Environment

- Ubuntu 20.04.4 LTS
- Python 3.9.12
- SQLAdmin 0.1.9

### Additional context

_No response_

### LINKED ISSUE #203: `str` is not Iterable
### Checklist

- [X] The bug is reproducible against the latest release or `master`.
- [X] There are no similar issues or pull requests to fix it yet.

### Describe the bug

As of 0.1.10, the templates use the new `is_iterable` function instead of `is_list` (#174). But now some list and details admin pages that have custom `column_formatters` or `column_formatters_detail` produce Internal Server Error.

It is because `str` object returned from the formatter is determined as `Iterable` and the formatted value is passed to `_url_for_details()`. It is a variation of #196.

```python
>>> from collections.abc import Iterable
>>> isinstance("foo", Iterable)
True
```

### Steps to reproduce the bug

_No response_

### Expected behavior

_No response_

### Actual behavior

_No response_

### Debugging material

```python
  File "/myproject/.venv/lib/python3.9/site-packages/sqladmin/templates/list.html", line 113, in block 'content'
    <a href="{{ model_admin._url_for_details(value) }}">({{ value }})</a>
  File "/myproject/.venv/lib/python3.9/site-packages/sqladmin/models.py", line 609, in _url_for_details
    pk = getattr(obj, inspect(obj).mapper.primary_key[0].name)
  File "/myproject/.venv/lib/python3.9/site-packages/sqlalchemy/inspection.py", line 71, in inspect
    raise exc.NoInspectionAvailable(
sqlalchemy.exc.NoInspectionAvailable: No inspection system is available for object of type <class 'str'>
```

### Environment

- Ubuntu 20.04.4 LTS
- Python 3.9.12
- SQLAdmin 0.1.9

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by okapies**
Fixes #196.
Fixes #203.

### CODE REVIEW

File: sqladmin/models.py [hunk 1]
```diff
@@ -734,25 +734,27 @@ def get_attr_value(
             result = getattr(obj, attr.key)
             result = result.value if isinstance(result, Enum) else result
 
-        return result or ""
+        return result
```
Comment (okapies) [inline sqladmin/models.py:737]: I found it also causes another bug. Is this an intentional change?

File: sqladmin/templates/details.html [hunk 2]
```diff
@@ -18,18 +18,19 @@ <h3 class="card-title">{{ model_admin.pk_column.name }}: {{ model_admin.get_attr
             {% for name, attr in model_admin._details_attrs %}
             <tr>
               <td>{{ name }}</td>
+              {% set value, formatted_value = model_admin.get_detail_value(model, attr) %}
               {% if (name, attr) in model_admin._details_relations %}
-              {% if is_iterable( model_admin.get_list_value(model, attr) ) %}
```
Comment (okapies) [inline sqladmin/templates/details.html:22]: I seems to be a typo of `get_detail_value(...)`, right?

File: sqladmin/models.py (hunk 1 above)
Comment (aminalaee) [reply sqladmin/models.py:737]: What issue did you face with this?

File: sqladmin/templates/details.html (hunk 2 above)
Comment (aminalaee) [reply sqladmin/templates/details.html:22]: Yes.

File: sqladmin/models.py [hunk 3]
```diff
@@ -734,25 +734,27 @@ def get_attr_value(
...
 
     def get_list_value(
         self, obj: type, attr: Union[Column, ColumnProperty, RelationshipProperty]
-    ) -> Any:
+    ) -> Tuple[Any, Any]:
         """Get instancee values for the list view."""
```
Comment (aminalaee) [inline sqladmin/models.py:742]:
```suggestion
        """Get tuple of (value, formatted_value) for the list view."""
```

File: sqladmin/models.py (hunk 1 above)
Comment (okapies) [reply sqladmin/models.py:737]: It returns `""` even if the `result` is an empty `InstrumentedList`, so it causes the following error:

```python
    def _url_for_details_with_attr(self, obj: Any, attr: RelationshipProperty) -> str:
        target = getattr(obj, attr.key)
        if target is None:
            return ""
    
>       pk = getattr(target, attr.mapper.primary_key[0].name)
E       AttributeError: 'InstrumentedList' object has no attribute 'id'
```

**Review by aminalaee — APPROVED** [review]
This looks great, thanks a lot @okapies .

**Comment by aminalaee** [comment]
BTW it would be great not to combine two issues in one PR, it makes both review and history cleaner. Thanks again 👍

**Pull request merged**